dftimewolf/lib/exporters/scp_ex.py

```diff
@@ -47,28 +49,47 @@ def SetUp(self, # pylint: disable=arguments-differ
     self._destination = destination
     self._hostname = hostname
     self._id_file = id_file
-    self._paths = paths.split(",")
+    if paths:
+      self._paths = paths.split(',')
+    else:
+      self._paths = None
     self._user = user
 
     if check_ssh and not self._SSHAvailable():
-      self.state.AddError("Unable to connect to host.", critical=True)
+      self.state.AddError('Unable to connect to host.', critical=True)
 
   def Process(self):
     """Copies the list of paths to the destination on user@hostname"""
+    if not self._paths:
+      fspaths = self.state.GetContainers(containers.FSPath)
+      self._paths = [fspath.path for fspath in fspaths]
+
+    if not self._paths:
+      self.state.AddError('No paths specified to SCP module.', critical=True)
+      return
+
     dest = self._destination
-    user = ""
+    user = ''
     if self._user:
-      user = "{0:s}@".format(self._user)
+      user = '{0:s}@'.format(self._user)
     if self._hostname:
-      dest = "{0:s}{1:s}:{2:s}".format(user, self._hostname, self._destination)
-    cmd = ["scp"]
+      dest = '{0:s}{1:s}:{2:s}'.format(user, self._hostname, self._destination)
+    cmd = ['scp']
     cmd.extend(self._paths)
     cmd.append(dest)
     print('Executing SCP command: {0:s}'.format(' '.join(cmd)))
     ret = subprocess.call(cmd)
     if ret != 0:
-      self.state.AddError("Failed copying {0!s}".format(self._paths),
+      self.state.AddError('Failed copying {0!s}'.format(self._paths),
                           critical=True)
+    else:
```

Comment:
It would be cool to change the check on `ret` to be a guard statement (replace this `else` by `return` and unindent the code below?)

Reply:
Done.